presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java

```diff
@@ -1422,6 +1423,11 @@ public SystemSessionProperties(
                         "The native engine executable file path for native engine execution",
                         featuresConfig.getNativeExecutionExecutablePath(),
                         false),
+                stringProperty(
+                        NATIVE_EXECUTION_PROGRAM_ARGUMENTS,
```

Comment:
@shrinidhijoshi is looking to allow single native process to be shared between multiple tasks / queries. Once we make that change, it would not be possible to control native process command args per-query. Hence, I wonder if we should not expose this flag via session properties and keep it configuration-only.

Reply:
That is a good point. But I'd like to think that being another refactoring when the shared process change is in. Because currently all c++ launching related parameters are passed in through session so we need to move them all together.